src/mlpack/methods/linear_svm/linear_svm.hpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file linear_svm.hpp
+ * @author Ayush Chamoli
+ *
+ * An implementation of Linear SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+#define MLPACK_METHODS_LINEAR_SVM_LINEAR_SVM_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <ensmallen.hpp>
+
+#include "linear_svm_function.hpp"
+
+namespace mlpack {
+namespace svm {
+
+/**
+ * The LinearSVM class implements an L2-regularized support vector machine
+ * model, and supports training with multiple optimizers and classification.
+ * The class supports different observation types via the MatType template
+ * parameter; for instance, support vector classification can be performed
+ * on sparse datasets by specifying arma::sp_mat as the MatType parameter.
+ *
+ * Linear SVM can be used for general classification tasks which will work
+ * on multiclass classification. More technical details about
+ * the model can be found on the following webpage:
+ *
+ * https://link.springer.com/content/pdf/10.1007%2FBF00994018.pdf
+ *
+ * @tparam MatType Type of data matrix.
+ */
+template <typename MatType = arma::mat>
+class LinearSVM
+{
+ public:
+  /**
+   * Construct the LinearSVM class with the provided data and labels.
+   * This will train the model. Optionally, the parameter 'lambda' can be
+   * passed, which controls the amount of L2-regularization in the objective
+   * function. By default, the model takes a small value.
+   *
+   * @tparam OptimizerType Desired differentiable separable optimizer
+   * @param data Input training features. Each column associate with one sample
+   * @param labels Labels associated with the feature data.
+   * @param numClasses Number of classes for classification.
+   * @param lambda L2-regularization constant.
+   * @param optimizer Desired optimizer.
+   */
+  template <typename OptimizerType>
+  LinearSVM(const MatType& data,
+            const arma::Row<size_t>& labels,
+            const size_t numClasses = 2,
+            const double lambda = 0.0001,
+            OptimizerType optimizer = OptimizerType());
+
+  /**
+   * Initialize the SoftmaxRegression without performing training.  Default
+   * value of lambda is 0.0001.  Be sure to use Train() before calling
+   * Classify() or ComputeAccuracy(), otherwise the results may be meaningless.
```

Comment:
You can also just add `lambda` as a parameter here, but I think it's okay either way.